launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/TranslationServlet.java

```diff
@@ -0,0 +1,108 @@
+  private static final String USER_FEED = "UserFeed";
+  private static final String RSS_FEED_LINK = "rssFeedLink";
+  private static final String LANGUAGE = "language";
+  private static final String XML_STRING = "xmlString";
+  private static final XmlMapper XML_MAPPER = new XmlMapper();
+  private static final String BASE_URL = "https://launchpod-step18-2020.appspot.com/rss-feed?id=";
+
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res)
+      throws JsonParseException, JsonMappingException, IOException {
+    String link = req.getParameter(RSS_FEED_LINK);
+    String targetLanguage = req.getParameter(LANGUAGE);
+    if (link == null || link == "") {
+      throw new IOException("Please give valid link.");
+    }
+    if (targetLanguage == null || targetLanguage == "") {
+      throw new IOException("Please give valid language.");
+    }
+    try {
+      XML_MAPPER.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+      String id = getIdFromUrl(link);
+      
+      //Search for key from given Link
+      Key desiredFeedKey = KeyFactory.stringToKey(id);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Entity desiredFeedEntity = datastore.get(desiredFeedKey);
+      String xmlString = (String) desiredFeedEntity.getProperty(XML_STRING);
+      RSS rssFeed = XML_MAPPER.readValue(xmlString, RSS.class);
+
+      // Translate fields to new language
+      Translate translate = TranslateOptions.getDefaultInstance().getService();
+
+      // Channel description
+      Translation translation = translate.translate(rssFeed.getChannel().getDescription(),
+          TranslateOption.targetLanguage(targetLanguage));
+      rssFeed.getChannel().setDescription(translation.getTranslatedText());
+
+      // Language
+      rssFeed.getChannel().setLanguage(targetLanguage);
+
+      // Episodes
+      for (Item item : rssFeed.getChannel().getItem()) {
+        // Episode title
+        translation = translate.translate(item.getTitle(), TranslateOption.targetLanguage(targetLanguage));
+        item.setTitle(translation.getTranslatedText());
+
+        // Episode description
+        translation = translate.translate(item.getDescription(), TranslateOption.targetLanguage(targetLanguage));
+        item.setDescription(translation.getTranslatedText());
+      }
+
+      // Generate Translated XML string then place it into datastore
+      String translatedXmlString = RSS.toXmlString(rssFeed);
+      Entity translatedUserFeedEntity = new Entity(USER_FEED);
+      translatedUserFeedEntity.setProperty(XML_STRING, translatedXmlString);
+      datastore.put(translatedUserFeedEntity);
+      String translatedFeedId = KeyFactory.keyToString(translatedUserFeedEntity.getKey());
+
+      // display new translated string to the user
+      res.setContentType("text/html");
+      res.getWriter().println(BASE_URL + translatedFeedId);
+
+    } catch (EntityNotFoundException e) {
```

Comment:
which line throws this exception? it's best practice to keep try-catch blocks as small as possible so it's clear what code threw the error